pallets/migrations/src/mock.rs

```diff
@@ -260,14 +260,19 @@ impl Default for ExtBuilder {
 
 impl ExtBuilder {
 	pub(crate) fn build(self) -> sp_io::TestExternalities {
-		let mut t = frame_system::GenesisConfig::default()
-			.build_storage::<Test>()
-			.expect("Frame system builds valid default genesis config");
-
-		GenesisBuild::<Test>::assimilate_storage(&pallet_migrations::GenesisConfig, &mut t)
-			.expect("Pallet migration's storage can be assimilated");
-
-		let mut ext = sp_io::TestExternalities::new(t);
+		let storage = frame_system::GenesisConfig::default()
+				.build_storage::<Test>()
+				.expect("Frame system builds valid default genesis config");
+	
+
+		// If we apply the genesis build, it will register in the storage the fact that all the
+		// migrations have already taken place, but we want to test that these migrations
+		// have executed.
+		// GenesisBuild::<Test>::assimilate_storage(&pallet_migrations::GenesisConfig, &mut storage)
+		//	.expect("Pallet migration's storage can be assimilated");
```

Comment:
Rather than disabling this entirely, we could extend the externality builder so that it has a field `prehistoric_migration` that defaults to empty but allows each test to individually populate it.

Reply:
Done, but with name `uncompleted_migrations` instead of `prehistoric_migration`